schema/entity-record.json

```diff
@@ -12,21 +12,129 @@
       "type": "boolean"
     },
     "entityType": {
+      "entityType": "object",
       "title": "Type",
-      "description": "From the entityType codelist. What kind of entity is this? The 'registeredEntity' code covers any legal entity created through an act of official registration, usually resulting in an identifier being assigned to the entity. The ‘legalEntity’ code covers other bodies with distinct legal personality (international institutions, statutory corporations etc.). The 'arrangement' code covers artificial entities, described in the data model for the purpose of associating one or more natural or legal persons together in an ownership or control relationship, but without implying that the parties to this arrangement have any other form of collective legal identity.",
-      "type": "string",
-      "enum": [
-        "registeredEntity",
-        "legalEntity",
-        "arrangement",
-        "anonymousEntity",
-        "unknownEntity",
-        "state",
-        "stateBody"
+      "description": "The type of entity described in the statement.",
+      "required": [
+        "type"
       ],
-      "codelist": "entityType.csv",
-      "openCodelist": false,
-      "propertyOrder": 4
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Entity type",
+          "description": "A value from the entityType codelist.",
+          "codelist": "entityType.csv",
+          "enum": [
+            "registeredEntity",
+            "legalEntity",
+            "arrangement",
+            "anonymousEntity",
+            "unknownEntity",
+            "state",
+            "stateBody"
+          ]
+        },
+        "subtype": {
+          "type": "string",
+          "title": "Entity subtype",
+          "description": "A value from the entitySubtype codelist. This MUST align with the `entityType` value.",
+          "codelist": "entitySubtype.csv",
+          "enum": [
+            "governmentDepartment",
+            "stateAgency",
+            "other",
+            "trust",
+            "nomination"
+          ],
+          "openCodelist": false
+        },
+        "details": {
+          "type": "string",
+          "title": "Details",
+          "description": "This may be used to provide a local name for this type of entity, or any further information to identify the type of entity. For example, in Finland 'ministeriö' for a government department."
+        }
+      },
+      "propertyOrder": 4,
+      "allOf":[
+        {
+          "if": {
+            "not":{
```

Comment:
I think the logic can be made a bit leaner by taking out this block from each of the allof subschemas:

```
        "not":{
          "properties":{
            "subtype":{
              "type":"null"
            }
          }
        }
 ```
...and putting it in a wrapping if statement. Like this: 


```
      "propertyOrder": 4,
      "if": {
        "not":{
          "properties":{
            "subtype":{
              "type":"null"
            }
          }
        }
      },
      "then": {
        "allOf":[
          {
            "if": {
              "properties": {
                "subtype": {
                  "const": "trust"
                }
              }
            },
            "then": {
              "properties": {
                "type":{
                  "enum":[
                    "arrangement",
                    "legalEntity"
                  ]
                }
              }
            }
          },
      [...]
```

You'd have to adjust test_schema.py again since 'null' would only appear the once.

Reply:
ooh yes this is much neater!!